packages/wp-source/src/libraries/handlers/__tests__/author.tests.ts

```diff
@@ -69,6 +69,40 @@ describe("author", () => {
     ]);
   });
 
+  test("overwrites the data when fetched with { force: true }", async () => {
+    // Add iniital data to the store
+    await store.libraries.source.populate({
+      state: store.state,
+      response: mockResponse(author1),
+    });
+    await store.actions.source.fetch("/author/author-1/");
+
+    // Mock Api responses
+    api.get = jest.fn().mockResolvedValueOnce(
+      mockResponse(
+        { ...author1, name: "Author 2" },
+        {
+          "X-WP-Total": "5",
+          "X-WP-TotalPages": "2",
+        }
+      )
+    );
+
+    expect((store.state.source.data["/author/author-1/"] as any).name).toEqual(
+      "Author 1"
+    );
```

Comment:
Uhm, I'll take a look.

Reply:
I found the problem. It's failing because `name` is not included in the `data` object but in the `author` object, so in order to get the name of the author you have to do

```ts
state.source.author[authorId].name
```

Remember that the author id is a number, not the link.

Reply:
Why is the `name` not included in the `data` though? Is this by design?

Reply:
If I remember correctly, the idea was to have in `data` only the props needed to get the entities from the state, along with other properties that are not found in entities (like which items belong to that specific category page, the number of pages, and so on).

For getting posts, pages, etc. you would need `type` and `id`.

For getting authors you just only need `id` so, yeah. I would say this is by design. :smile: